### ISSUE TITLE
Support collapse-able sections in XML and JSON

### ISSUE THREAD

**Issue by ElliotSilver** [HL7/fhir-ig-publisher#835]
It would be nice to be able to collapse the narrative section when looking at the JSON or XML of a definitional resource in an IG. Similarly, I often want to skip over the snapshot, and go straight to the differential; collapsing the snapshot would be helpful.

**Comment by grahamegrieve** [comment]
does this same idea apply anywhere else but profiles?

**Comment by ElliotSilver** [comment]
Yes, any resource that has a large narrative, it would be nice to collapse to get to the actual content. Also on any resource that has a large element or significant repeating elements. For example, collapsing the expansion of a ValueSet in order to get to the definition.

**Comment by costateixeira** [comment]
if we turn off wrapping and enable scroll bars, the narrative will not be
taking all the space it currently takes. See here:
https://chat.fhir.org/#narrow/stream/196008-ig-publishing-requirements/topic/JSON.20.28.26.20XML.29.20rendering/near/410765323

**Comment by lmckenzi** [comment]
We will add an ability to the publisher that when spitting out the XML, JSON, or TTL HTML views to hard-code a per-resource list of "key elements" where the tooling will check to see "is this element present?" and if so, will (at the top of the fragment) include a "jump to ..." list with hyperlinks to those locations, and anchors embedded at the appropriate point.  E.g. "Jump to [differential] [snapshot]" for a StructureDefinition.
